library/src/test/java/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModelTest.kt

```diff
@@ -136,4 +159,40 @@ class MessageListViewModelTest {
 
         verify(client).flag(MESSAGE.user.id)
     }
+
+    @Test
+    fun `Should return from thread to normal mode on back click`() {
+        val viewModel = MessageListViewModel(CID, domain, client)
+        viewModel.onEvent(MessageListViewModel.Event.ThreadModeEntered(MESSAGE))
+        val states = viewModel.state.observeAll()
+
+        viewModel.onEvent(MessageListViewModel.Event.BackButtonPressed)
+
+        states.last() shouldBeInstanceOf  MessageListViewModel.State.Result::class
```

Comment:
`MessageListViewModel.State.Result` is a data class, and it helps us generating `.equals()` method and so on.
Would be great to check, not only that the last state is instance of this class, but also that the content of the result is the one we expect.
You can create a variable `expectedState` with the expected result and update the assertion with
```
states.last `should be equal to` expectedState
```

Reply:
This is not possible (in reasonable time) to define complete, expected instances of `MessageListViewModel.State.Result`. However, I did add more specific assertions to validate all the data that are supposed to be tested here.

Reply:
(Classes wrapped by `Result` are too complex)

Reply:
Ok, no problem, I didn't check the complexity of the wrapped class.
As we improve our `Mother` class, the easier it will be to create these values for us